src/fragments/lib/auth/common/mfa/flows.mdx

```diff
@@ -0,0 +1,214 @@
+
+Therefore it is recommended to use the `updateMFAPreference` API to set the MFA preferences for the user, if the user decides to set up any MFA methods. An example use case would be that when a user successfully sets up TOTP using the `setupTOTP` and `verifyTOTPSetup` API, `updateMFAPreference` should be invoked to set the MFA preferences for the user i.e. marking TOTP as enabled.
+
+</Callout>
+
+### Adding phone number during Sign Up
+
+Sign up users normally with the chosen `Username` type and password. `"phone_number"` will be required in the `userAttributes` if SMS MFA is ON, or manually enabled for the user. Note that if the username attribute is the `phone_number`, then adding a separate attribute for phone number is not required during sign up.
+
+import flutter6 from "/src/fragments/lib/auth/flutter/sms/sign_up.mdx";
+
+<Fragments fragments={{flutter: flutter6}} />
+
+import iosSignUp from "/src/fragments/lib/auth/ios/sms/sign_up.mdx";
+
+<Fragments fragments={{ios: iosSignUp}} />
+
+Verification of user accounts is done via the `confirmSignUp` method with the OTP sent to their phone or email.
+
+import flutter7 from "/src/fragments/lib/auth/flutter/sms/confirm_sign_up.mdx";
+
+<Fragments fragments={{flutter: flutter7}} />
+
+import iosConfirmSignUp from "/src/fragments/lib/auth/ios/sms/confirm_sign_up.mdx";
+
+<Fragments fragments={{ios: iosConfirmSignUp}} />
+
+### Handling SMS MFA challenge during Sign In
+
+Sign in users normally with the chosen `Username` type and password.
+
+import flutter8 from "/src/fragments/lib/auth/flutter/sms/sign_in.mdx";
+
+<Fragments fragments={{flutter: flutter8}} />
+
+import iosSignInTOTP from "/src/fragments/lib/auth/ios/sms/sign_in.mdx";
+
+<Fragments fragments={{ios: iosSignInTOTP}} />
+
+If MFA is **ON** or enabled for the user, you must call `confirmSignIn` with the OTP sent to their phone.
+
+import flutter9 from "/src/fragments/lib/auth/flutter/sms/confirm_sign_in.mdx";
+
+<Fragments fragments={{flutter: flutter9}} />
+
+import iosConfirmSignInTOTP from "/src/fragments/lib/auth/ios/sms/confirm_sign_in.mdx";
+
+<Fragments fragments={{ios: iosConfirmSignInTOTP}} />
+
+## Multi-factor authentication with TOTP
+
+You can use Time-based One-Time Password (TOTP) for multi-factor authentication (MFA) in your web or mobile applications. Amplify's Auth category includes support for handling TOTP setup and codes, offering an integrated solution for enhanced security for your users. Authenticator apps generate the temporary, one-time passwords. These apps, such as Google Authenticator, Microsoft Authenticator, have the TOTP algorithm built-in and work by using a shared secret key and the current time to generate a short-lived password.
```

Comment:
Small nits:

You can use Time-based One-Time Password (TOTP) for multi-factor authentication (MFA) in your web or mobile applications. The Amplify Auth category includes support for handling TOTP setup and verification using authenticator apps, offering an integrated solution for enhanced security for your users. These apps, such as Google Authenticator, Microsoft Authenticator, have the TOTP algorithm built-in and work by using a shared secret key and the current time to generate a short-lived password.